pyscriptjs/src/exceptions.ts

```diff
@@ -78,25 +66,21 @@ export function _createAlertBanner(
             break;
     }
 
-    const banner = document.createElement('div');
-    banner.className = `alert-banner py-${level}`;
-
-    if (messageType === 'html') {
-        banner.innerHTML = message;
-    } else {
-        banner.textContent = message;
-    }
+    const content = messageType === 'html' ? 'innerHTML' : 'textContent';
+    const banner = Object.assign(document.createElement('div'), {
+        className: `alert-banner py-${level}`,
+        [content]: message,
+    });
 
     if (level === 'warning') {
-        const closeButton = document.createElement('button');
+        const closeButton = Object.assign(document.createElement('button'), {
+            id: 'alert-close-button',
+            innerHTML: CLOSEBUTTON,
+        });
```

Comment:
I haven't seen this logic before, does this just create the object returned from `createElement` with some extra attributes (id, innerHTML?)

This seems like a cool trick!

Reply:
[Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) returns the `target` (first argument) after attaching fields/properties via the object(s) that follow such target. In this specific case it makes multiple fields attached at once so that browsers/engine can also optimize when/if necessary in one shot, instead of two or more.